sm2/Cargo.toml

```diff
@@ -18,24 +18,30 @@ edition = "2024"
 rust-version = "1.85"
 
 [dependencies]
-elliptic-curve = { version = "0.14.0-rc.0", default-features = false, features = ["sec1"] }
+elliptic-curve = { version = "0.14.0-rc.0", default-features = false, features = [
+    "sec1",
+] }
 rand_core = { version = "0.9", default-features = false }
 
 # optional dependencies
 primeorder = { version = "=0.14.0-pre.2", optional = true, path = "../primeorder" }
 rfc6979 = { version = "=0.5.0-pre.4", optional = true }
 serdect = { version = "0.3", optional = true, default-features = false }
-signature = { version = "=2.3.0-pre.6", optional = true, features = ["rand_core"] }
+signature = { version = "=2.3.0-pre.6", optional = true, features = [
+    "rand_core",
+    "digest",
+] }
 sm3 = { version = "=0.5.0-pre.5", optional = true, default-features = false }
 
 [dev-dependencies]
 hex-literal = "1"
 proptest = "1"
 rand_core = { version = "0.9", features = ["os_rng"] }
+rand = "0.9"
```

Comment:
It seems like you're adding this to change the documentation from using `OsRng` to `rand::rng`.

IMO `OsRng` is the safer option for things like key generation. `rand::rng` can use a userspace RNG which can introduce problems like fork safety.

Reply:
Sidebar: we should also add APIs that encapsulate this so the user doesn't need to pass an explicit RNG at all